test/python/circuit/test_singleton.py

```diff
@@ -325,20 +326,42 @@ def __init__(self, x):
         self.assertEqual(gate.x, 1)
         self.assertIsNot(MyAbstractGate(1), MyAbstractGate(1))
 
-    def test_inherit_singleton(self):
-        class Measure(SingletonInstruction):
-            def __init__(self):
-                super().__init__("measure", 1, 1, [])
+    def test_return_type_singleton_instructions(self):
+        measure = Measure()
+        self.assertEqual(measure.__class__.__name__, "_SingletonMeasure")
+
+        reset = Reset()
+        self.assertEqual(reset.__class__.__name__, "_SingletonReset")
```

Comment:
This test isn't quite right - technically it'll pass at the moment, but it's reliant on private internal details of the logic.  A singleton is an object which is literally _the same object_ on each return - in Python, that means `a is b`.  You can still the style of `assertIs` tests above for how we test singletons.